apps/dcos_net/src/dcos_net_mesos_listener.erl

```diff
@@ -997,3 +1011,55 @@ stream(Data, #state{pid=Pid, size=Size, buf=Buf}=State) ->
             httpc:stream_next(Pid),
             {next, State#state{buf=Buf0}}
     end.
+
+%%%===================================================================
+%%% Metrics functions
+%%%===================================================================
+
+-spec(init_metrics() -> ok).
+init_metrics() ->
+    prometheus_summary:new([
+        {registry, mesos_listener},
+        {name, pubsub_duration_seconds},
+        {labels, []},
+        {help, "Time spent processing pubsub"}]), % improve on this description
+    prometheus_counter:new([
+        {registry, mesos_listener},
+        {name, bytes_total},
+        {labels, []},
+        {help, "Total number of bytes received"}]),
+    prometheus_counter:new([
+        {registry, mesos_listener},
+        {name, failures_total},
+        {labels, []},
+        {help, "Total number of bytes failures"}]),
+    prometheus_counter:new([
+        {registry, mesos_listener},
+        {name, redirects_total},
```

Comment:
I think this metric should have a better name, a help string or/and even a type (i.e. boolean `standby`?)

Reply:
There is something here I do not quite understand. Is the goal of the `redirects_total` metric to show in which moments of time are we being redirected? Do we plan to use it knowing that a redirect means that we are not talking to a master?

Reply:
Prometheus has native support for boolean values. it is supported in `prometheus.erl` library as well [1]. Just add it to the beginning of the list of all collectors [2].

[1] https://github.com/deadtrickster/prometheus.erl/blob/v4.2.2/src/metrics/prometheus_boolean.erl
[2] https://github.com/dcos/dcos-net/blob/b30888b/config/sys.config#L98